tests/integration/helpers_test/test_helpers_utils.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from utils import override_config
+
+
+def test_override_config(app):
+    with override_config(foo1='bar1', foo2='bar2'):
+        from flask import current_app
+        assert current_app.config['foo1'] == 'bar1'
+        assert current_app.config['foo2'] == 'bar2'
+
+    assert not hasattr(current_app.config, 'foo1')
+    assert not hasattr(current_app.config, 'foo2')
```

Comment:
If there was a `leak` this doesn't detect it, I think you meant to use `not in`:
```python
In [1]: config = {'foo1'}

In [2]: not hasattr(config, 'foo1')
Out[2]: True

In [3]: not hasattr(config, 'not there')
Out[3]: True

In [4]: 'foo1' not in config
Out[4]: False

In [5]: 'not there' not in config
Out[5]: True
```
Also maybe it would be worth to check if the stuff that was in the config before is also available when overriding?

Reply:
Oh yes, nice points both of them, I'll do that. I overlooked `current_app.config` and thought it was an object instead of a dict.

Reply:
Done